implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java

```diff
@@ -244,26 +262,39 @@ Metric writeHistogramSupport(HistogramSupport histogramSupport) {
                 histogramDataPoint.addBucketCounts((long) countAtBucket.count());
             }
             metricBuilder.setHistogram(Histogram.newBuilder()
-                    .setAggregationTemporality(AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE)
+                    .setAggregationTemporality(AggregationTemporality.forNumber(config.getAggregationTemporality()))
                     .addDataPoints(histogramDataPoint));
             return metricBuilder.build();
         }
 
         return metricBuilder.setHistogram(Histogram.newBuilder()
-                .setAggregationTemporality(AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE)
+                .setAggregationTemporality(AggregationTemporality.forNumber(config.getAggregationTemporality()))
                 .addDataPoints(histogramDataPoint)).build();
     }
 
     // VisibleForTesting
     Metric writeFunctionTimer(FunctionTimer functionTimer) {
         return getMetricBuilder(functionTimer.getId()).setHistogram(Histogram.newBuilder()
                 .addDataPoints(HistogramDataPoint.newBuilder().addAllAttributes(getTagsForId(functionTimer.getId()))
-                        .setStartTimeUnixNano(((StartTimeAwareMeter) functionTimer).getStartTimeNanos())
-                        .setTimeUnixNano(TimeUnit.MILLISECONDS.toNanos(this.clock.wallTime()))
+                        .setStartTimeUnixNano(getStartTimeNanos((functionTimer))).setTimeUnixNano(getEndTimeNanos())
                         .setSum(functionTimer.totalTime(getBaseTimeUnit())).setCount((long) functionTimer.count())))
                 .build();
     }
 
+    private boolean isDeltaAggregationTemporality() {
+        return config.getAggregationTemporality() == AggregationTemporality.AGGREGATION_TEMPORALITY_DELTA_VALUE;
+    }
+
+    private long getStartTimeNanos(Meter meter) {
```

Comment:
I agree from the calculation point of view that we will have 2 calls to "clock.wallTime()". But since we are trying to normalize the steps, the final values would have much difference and will point to the same step start and end times.

Reply:
Yeah, these are the things we ned to think through and also test since we can bump into corner cases/tricky scenarios, right now, it makes sense to me but haven't tried to do anything "mean" to it. :D

Reply:
Added additional test cases for this.